### ISSUE TITLE
When Oh My Zsh is sourced, insert-unambiguous does not work when fzf completion is also enabled 

### ISSUE THREAD

**Issue by paw-lu** [marlonrichert/zsh-autocomplete#118]
# Environment
* Zsh version: **5.8**
* `zsh-autocomplete` version: **4aa819f98e0577**

# Steps to reproduce

Please report if the bug occurs without config files:
```zsh
$ cd $(mktemp -d)  # Create a temp dir and enter it.
$ ZDOTDIR=$PWD HOME=$PWD zsh -f  # Start a subshell in it without config files.
% touch prefix_{1..7}.txt  # Create seven files with identical prefixes to test partial completion
% source /Users/pawlu/.oh-my-zsh/custom/plugins/zsh-autocomplete/zsh-autocomplete.plugin.zsh  # Source the plugin.
% zstyle ':autocomplete:tab:*' insert-unambiguous yes
% zstyle ':autocomplete:tab:*' fzf yes
% [ -f /Users/pawlu/.fzf.zsh ] && source /Users/pawlu/.fzf.zsh  # Source fzf
% source /Users/pawlu/.oh-my-zsh/oh-my-zsh.sh  # Source Oh My Zsh
% cat pre<TAB>  # Try to invoke partial completion 
% cat prefix_1.txt  # Entirety of one file is completed
```
Does the bug occur without config files?
* [x] Yes
* [ ] No

## Expected behavior
I expected the completion to insert the prefix, completing it to

```sh
% cat prefix_
```

## Actual behavior
I entirety of the first file name was used in the completion, completing it to

```sh
% cat prefix_1.txt
```

Related to question on #82

**Comment by marlonrichert** [comment]
Fixed, but please note that if you're going to `source` OMZ from the command line, it's still going to break. There's nothing I can do about that. But if you `source` everything in your `~/.zshrc` file, then it should all be fine.

**Comment by paw-lu** [comment]
@marlonrichert I tried this again, but it still does not seem to work for me.

Using a `.zshrc` this time to avoid the issue you mentioned.

```sh
export ZSH=$HOME/.oh-my-zsh
plugins = (
    zsh-autocomplete
    zsh-autosuggestions
)
zstyle ':autocomplete:tab:*' insert-unambiguous yes
zstyle ':autocomplete:tab:*' fzf yes
source $ZSH/oh-my-zsh.sh
```

```sh
% cd $(mktemp -d)
% touch prefix_{1..3}.txt
% cat pre<TAB>
% cat prefix_1.txt
```

**Comment by marlonrichert** [comment]
This time, it was actually a different problem than the last time, but it's fixed now, too.

**Comment by paw-lu** [comment]
@marlonrichert can confirm!

Thank you so much for both this project and the change!

**Comment by rwmitchell** [comment]
This looks like my issue, when I press TAB, the first match is added to the cmdline, not just the portion that is shared with all matches.

These are my settings:
```zstyle ':autocomplete:*' fzf-completion yes
zstyle ':autocomplete:*' insert-unambiguous yes
zstyle ':autocomplete:*' widget-style menu-select
zstyle ':autocomplete:*' min-delay 1.0
zstyle ':autocomplete:*' min-input 3

source $ZSH/custom/plugins/zsh-autocomplete/zsh-autocomplete.plugin.zsh```

autocomplete version: 185aa653e66cb10bb6ddd39c45a260422f9a036a
zsh: zsh-5.8-239-g8becb89
OS: darwin18.7.0

**Comment by rwmitchell** [comment]
recently added fzf and suddenly TAB is working like I expected/described.  go figure.

**Comment by marlonrichert** [comment]
Well yeah, `zstyle ':autocomplete:*' fzf-completion yes` won't work without `fzf`. It even says so here: 
https://github.com/marlonrichert/zsh-autocomplete/blob/c970b6d16af808e7bdef3f7094f4cc5c81fe0c10/.zshrc#L50
https://github.com/marlonrichert/zsh-autocomplete/blob/c970b6d16af808e7bdef3f7094f4cc5c81fe0c10/.zshrc#L51

**Comment by rwmitchell** [comment]
certainly that part won't and I had tried disabling it (after the post above), but I was still getting the first menu item filled in on the cmdline even though there were many other options with zero letters in common - it was that part I was trying to fix.  I wanted TAB to show possible completions, but not change the cmdline - at least  not until the second TAB hit (where it normally starts cycling thru the completion menu).  Instead of it was auto putting the first entry on the cmdline.

**Comment by rwmitchell** [comment]
Ah, I had changed what I was doing.  ```ls z[TAB]``` expands filenames starting with z until they don't match (goal behavior)
typing just ```ls [TAB]``` still fills in the first file in directory.

**Comment by marlonrichert** [comment]
I did find a pair of bugs in the common substring logic. They're fixed now.

---

PS: If you enjoy using my software, please consider [sponsoring me](https://github.com/sponsors/marlonrichert). 🙂

**Comment by rwmitchell** [comment]
hmmm, ```ls z[TAB]``` is putting the first entire  item on the cmdline instead of only filling to the first unambiguous character.
```ls [TAB]``` still puts the first item on the cmdline, even if it shares nothing in common with the other items.

**Comment by marlonrichert** [comment]
@rwmitchell Instead of commenting  on a closed issue that is not actually related to your problem, can you please open a new issue for this? Thanks.

**Comment by marlonrichert** [comment]
OK, it turns out there was another bug in the "unambiguous" completion that my own config didn't happen to hit but was trigger by yours. Sorry about that. I will push in a fix later today.

**Comment by marlonrichert** [comment]
Fixed.
